server/src/main/java/org/opensearch/action/admin/indices/tiering/TransportHotToWarmTieringAction.java

```diff
@@ -90,21 +97,34 @@ protected void clusterManagerOperation(
         ClusterState state,
         ActionListener<HotToWarmTieringResponse> listener
     ) throws Exception {
-        Index[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
+        Index[] concreteIndices = indexNameExpressionResolver.concreteIndicesWithTier(state, request, IndexModule.TieringState.HOT);
         if (concreteIndices == null || concreteIndices.length == 0) {
+            logger.info(
+                "[HotToWarmTiering] No hot concrete indices resolved for the indices {} in the request",
+                List.of(request.indices())
+            );
             listener.onResponse(new HotToWarmTieringResponse(true));
             return;
         }
         final TieringValidationResult tieringValidationResult = validateHotToWarm(
             state,
-            Set.of(concreteIndices),
+            newHashSet(concreteIndices),
             clusterInfoService.getClusterInfo(),
             diskThresholdSettings
         );
 
         if (tieringValidationResult.getAcceptedIndices().isEmpty()) {
-            listener.onResponse(tieringValidationResult.constructResponse());
+            listener.onResponse(constructToHotToWarmTieringResponse(tieringValidationResult.getRejectedIndices()));
             return;
         }
+
+        final TieringUpdateClusterStateRequest updateClusterStateRequest = new TieringUpdateClusterStateRequest(
+            tieringValidationResult.getRejectedIndices(),
+            request.waitForCompletion()
+        ).ackTimeout(request.timeout())
```

Comment:
where is this `ackTimeout` used ?

Reply:
refer: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/ack/ClusterStateUpdateRequest.java#L53